addons/stock/tests/test_packing.py

```diff
@@ -1890,6 +1890,67 @@ def test_pick_another_pack(self):
             {'package_id': pack3.id, 'state': 'assigned', 'is_done': True},
         ])
 
+    def test_reusable_package_weight(self):
+        reusable_box = self.env['stock.quant.package'].create({
+            'name': 'Reusable Box',
+            'package_use': 'reusable',
+        })
+
+        delivery_a = self.env['stock.picking'].create({
+            'picking_type_id': self.warehouse.out_type_id.id,
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'move_ids_without_package': [Command.create({
+                'name': self.productA.name,
+                'product_id': self.productA.id,
+                'product_uom_qty': 5.0,
+                'location_id': self.stock_location.id,
+                'product_uom': self.productA.uom_id.id,
+                'location_dest_id': self.customer_location.id,
+            })],
+            'move_line_ids': [Command.create({
+                'location_id': self.stock_location.id,
+                'result_package_id': reusable_box.id,
+                'quantity': 5.0,
+                'product_id': self.productA.id,
+                'location_dest_id': self.customer_location.id,
+                'product_uom_id': self.productA.uom_id.id,
+            })],
+        })
+        reusable_box.shipping_weight = 5
```

Comment:
```suggestion
```
Instead of manually setting the package's `shipping_weight` (which kinda truncates the computation, as it will directly use it), I think it would be more useful to set a `weight` to the product and work with computed values.